sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc

```diff
@@ -284,17 +318,176 @@ a description of the layouts and detailed layout-specific rules. +
 'Stride' further qualifies how matrix elements are laid out in memory. It must be a
 scalar 'integer type' and its exact semantics depend on 'MemoryLayout'. +
  +
+'Memory Operand', if present, must begin with a _Memory Operand_ literal.
+If not present, it is the same as specifying the _Memory Operand_ None. +
+ +
+All the operands to this instruction must be dynamically uniform within every
+instance of the 'Scope' of the cooperative matrix. +
+ +
 1+|Capability: +
 *{capability_prefetch_name}*
-1+| 8+variable | {OpCooperativeMatrixPrefetchINTEL_token} | '<id>' +
+1+| 6+variable | {OpCooperativeMatrixPrefetchINTEL_token} | '<id>' +
 'Pointer' | '<id>' +
-'X offset' | '<id>' +
-'Y offset' | '<id>' +
 'Rows' | '<id>' +
 'Columns' | Literal +
 'Cache Level' | '<id>' +
 'MemoryLayout' | Optional '<id>' +
-'Stride' |
+'Stride' | Optional +
+'Memory Operand' |
+|=====
+
+[cols="1,1,10*3",width="100%"]
+|=====
+11+|[[OpCooperativeMatrixLoadCheckedINTEL]]*OpCooperativeMatrixLoadCheckedINTEL* +
+ +
+Load a cooperative matrix through a pointer. Global matrix size might be not multiple the size of
+the two-dimentional region that is being loaded, in this case the out-of-bounds elements are
+set to 0. +
+ +
+'Result Type' is the type of the loaded object. It must be a cooperative matrix
+type. +
+ +
+'X offset' must be a scalar 32-bit integer type. It specifies offset in number of elements
+along X axis from the 'Pointer' where the loaded memory region starts from. +
+ +
+'Y offset' must be a scalar 32-bit integer type. It specifies offset in number of elements
+along Y axis from the 'Pointer' where the loaded memory region starts from. +
+ +
+'Pointer' is a pointer. Its type must be an *OpTypePointer* whose 'Type' operand
+is a scalar or vector type. If the *Shader* capability was declared, 'Pointer'
+must point into an array and any *ArrayStride* decoration on 'Pointer' is ignored. +
+ +
+'MemoryLayout' specifies how matrix elements are laid out in memory. It must come
+from a 32-bit integer 'constant instruction' whose value corresponds to a
+'Cooperative Matrix Layout'. See the _Cooperative Matrix Layout_ table for
+a description of the layouts and detailed layout-specific rules. +
+ +
+'Height' is the height (number of rows of a big matrix) of the two-dimensional
+region to load the matrix from. It must be a scalar 'integer type'. +
+ +
+'Width' is the width (number of columns of a big matrix) of the two-dimensional
+region to load the matrix from. It must be a scalar 'integer type'. +
```

Comment:
We should describe the units for the width, specifically whether it is in units of bytes or elements (I think it's bytes?).